### PR TITLE
[XLA] Make replay_computation accept input HLO file with multiple module.

### PR DESCRIPTION
**Pull request by nouiz**
I needed that today.
It allow to use replay_computation with hlo tests file.

This use the same syntax as hlo_to_llvm_ir use.

@sanjoy

### CODE REVIEW

File: tensorflow/compiler/xla/tools/replay_computation.cc [hunk 1]
```diff
@@ -400,21 +400,31 @@ StatusOr<HloSnapshot> ParseSingleHloFile(const string& filename,
...
+    StatusOr<std::unique_ptr<HloModule>> module =
+      ParseAndReturnUnverifiedModule(hlo_module_text, config);
+    if (module.ok()) {
+      HloSnapshot snapshot;
+      *snapshot.mutable_hlo()->mutable_hlo_module() =
         module.ValueOrDie()->ToProto();
-    return snapshot;
-  } else {
-    LOG(ERROR) << module.status();
+      //return std::vector<HloSnapshot>{std::move(snapshot)};
```
Comment (sanjoy) [inline tensorflow/compiler/xla/tools/replay_computation.cc:420]: Stray comment?

**Review by sanjoy — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/tools/replay_computation.cc (hunk 1 above)
Comment (nouiz) [reply tensorflow/compiler/xla/tools/replay_computation.cc:420]: I amended the commit to remove that line.

**Comment by cheshire** [comment]
I’d really prefer not to use the “---” trick, could we not use xargs?

**Comment by nouiz** [comment]
@cheshire

> I’d really prefer not to use the “---” trick, could we not use xargs?

The reason I understood for this request is that the error report are more difficult to understand due to the line numbers being wrong. I also got hit by this, but by using unique hlo module name, it was a little bit better.
Is there other reasons you do not like workflow?

Having tons of very small files, even in organized directory, isn't a great alternative to packing many small hlo in the same file.

What about replay_computation printing the line number offset before or after the FileCheck error?
It could also print the number of HLO passing before the one that error out.
That will help find where the error come from.

If you agree with that, I can implement them easily in this PR.

**Comment by nouiz** [comment]
I updated this PR with a commit that print the line where the section start.
This should resolve the hard to understand line number issue you raised.

**Review by cheshire — APPROVED** [review]

**Pull request merged**